client/src/view/global-fabric-panel.tsx

```diff
@@ -72,50 +64,56 @@ export function GlobalFabricPanel({constructFabric, fabric, cancelSelection}: {
-                        </DropdownToggle>
-                        <DropdownMenu>
-                            {loadFabricCode().map((code, index) => (
-                                <DropdownItem key={`Buffer${index}`} onClick={() => select(code, index)}>
-                                    {code}
-                                </DropdownItem>
-                            ))}
-                        </DropdownMenu>
-                    </ButtonDropdown>
-                </ButtonGroup>
-                <ButtonGroup className="mx-3">
-                    <Button onClick={() => constructFabric(loadFabricCode()[storageIndex])}><FaRecycle/></Button>
-                    <Button onClick={() => withFabric(f => {
-                        const dateString = new Date().toISOString()
-                            .replace(/[.].*/, "").replace(/[:T_]/g, "-")
-                        const output = f.output
-                        FileSaver.saveAs(extractJointBlob(output), `${dateString}-joints.csv`)
-                        FileSaver.saveAs(extractIntervalBlob(output), `${dateString}-intervals.csv`)
-                    })}><FaDownload/></Button>
-                </ButtonGroup>
-                <ButtonGroup className="mx-3">
-                    <Button onClick={() => withFabric(f => f.optimize(false))}><FaBolt/>L</Button>
-                    <Button onClick={() => withFabric(f => f.optimize(true))}><FaBolt/>H</Button>
-                    <Button onClick={() => withFabric(connectClosestFacePair)}><FaStarOfDavid/></Button>
-                </ButtonGroup>
-                <ButtonGroup className="mx-3">
-                    <Button onClick={() => withFabric(f => f.instance.setAltitude(10))}><FaParachuteBox/></Button>
-                    <Button onClick={() => withFabric(f => f.autoRotate = !f.autoRotate)}><FaSyncAlt/></Button>
-                    <Button onClick={() => withFabric(f => f.instance.centralize())}><FaCompressArrowsAlt/></Button>
-                </ButtonGroup>
-            </ButtonToolbar>
+        <div className="text-center">
+            <ButtonGroup className="w-75 align-self-center my-4" vertical={true}>
+                <ButtonDropdown className="w-100 my-2 btn-info" isOpen={open} toggle={() => setOpen(!open)}>
+                    <DropdownToggle>
+                        {open ? <FaRegFolderOpen/> : <FaRegFolder/>} Choose a program
+                    </DropdownToggle>
+                    <DropdownMenu>
+                        {loadFabricCode().map((code, index) => (
+                            <DropdownItem key={`Buffer${index}`} onClick={() => select(code, index)}>
+                                {code}
+                            </DropdownItem>
+                        ))}
+                    </DropdownMenu>
+                </ButtonDropdown>
+                <Button className={buttonClass} onClick={() => withFabric(f => f.optimize(false))}>
+                    <FaBolt/> L Optimize
```

Comment:
Can you make this more descriptive than "L Optimize"? I have no idea what it means.

Reply:
Haha, yes how to give a name to something I couldn't describe easily to you. I've renamed a bit, though.